### PR TITLE
refactor(service): wireguard repos updated

### PR DESCRIPTION
**Pull request by voryzen**
Wireguard repos, updated.
Wireguard installs, updated

Added options for debian, and CentOS

### CODE REVIEW

**Comment by voryzen** [comment]
@robjloranger

I've made a PR for you to test.

File: roles/vivumlab_wireguard/tasks/main.yml [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10")
 
-- name: Install WireGuard
+- name: Add the WireGuard PPA repository (Debian)
+  apt_repository:
+    repo: deb http://deb.debian.org/debian/ unstable main
+    state: present
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] !== "10" and ansible_facts[distribution_major_version] !== "11")
```
Comment (denis-ev) [inline roles/vivumlab_wireguard/tasks/main.yml:14]:
```suggestion
  when: (ansible_facts['distribution'] == "Debian") and (ansible_facts[distribution_major_version] != "10" or ansible_facts[distribution_major_version] != "11")
```
Did you test this? Unequal usually looks like this `!=`.

File: roles/vivumlab_wireguard/tasks/main.yml [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
   apt_repository:
-    repo: ppa:wireguard/wireguard
+    repo: deb http://deb.debian.org/debian buster-backports main
     state: present
-  register: result
-  until: result is succeeded
-  retries: 10
-  delay: 3
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10")
```
Comment (denis-ev) [inline roles/vivumlab_wireguard/tasks/main.yml:7]:
```suggestion
  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10") or (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "11")
```
You excluded Version 11 in the other one, shouldn't this include 11 then? Just a guess, I'm usually using Ubuntu.

File: roles/vivumlab_wireguard/tasks/main.yml [hunk 3]
```diff
@@ -1,19 +1,37 @@
 ---
-- name: Add the WireGuard PPA repository
+- name: Add the WireGuard PPA repository (Debian)
```
Comment (denis-ev) [inline roles/vivumlab_wireguard/tasks/main.yml:2]:
```suggestion
- name: Add the buster-backports repository (Debian)
```

**Review by denis-ev — APPROVED** [review]

File: roles/vivumlab_wireguard/tasks/main.yml (hunk 2 above)
Comment (voryzen) [reply roles/vivumlab_wireguard/tasks/main.yml:7]: For buster backports? No
Buster-backports are programs/files whatever the exist in the new er version, but might be needed in a previous version.
So they 'backport' to 'buster'.

These files exist in bullseye (Version 11 of debian) already.

**Comment by voryzen** [comment]
This should be ready to merge.

**Review by denis-ev — APPROVED** [review]

**Pull request merged**